### PR TITLE
[ML] [Data Frame] adding and modifying auditor messages

### LINKED ISSUE #42717: [ML] Missing data frame notifications for create, stop and delete
Current notifications for a single data frame batch run as follows, according to 7.2.0-BC2:
image

Data frame notifications are missing for 
- created
- stopped
- deleted

In addition, it might be useful to know it was a pivot transform.

The first message also sounds like an error, when it is going to be happy path. Change from `Could not find destination index [df-barcelona-district]. Creating index with deduced mappings.`

to 

`Creating destination index [df-barcelona-district] with deduced mappings.`

_Once we run continuously, then we should revisit if each checkpoint should have a message, and if we want any stats in there (e.g. checkpoint 123, 15000000 docs indexed)_

### PR DESCRIPTION
**Pull request by benwtrent**
Adds new audit messages for create and delete transform. Addresses the verbiage of a couple of messages, and changes some references from the `indexer` (an internal concept) to `data frame transform`

closes #42717

blocked by: #42644

### CODE REVIEW

**Comment by benwtrent** [comment]
run elasticsearch-ci/bwc

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java [hunk 1]
```diff
@@ -553,12 +553,15 @@ protected void doSaveState(IndexerState indexerState, Map<String, Object> positi
                                     },
                                     statsExc -> {
                                         logger.error("Updating stats of transform [" + transformConfig.getId() + "] failed", statsExc);
+                                        auditor.warning(getJobId(),
+                                            "Failure updating persisting stats of transform: " + statsExc.getMessage());
```
Comment (droberts195) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java]: Maybe remove the word "persisting", i.e. `updating persisting stats` -> `updating stats`?

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
run elasticsearch-ci/bwc

**Pull request merged**